versioned/storage/rocksdb/src/main/java/org/projectnessie/versioned/storage/rocksdb/RocksDBPersist.java

```diff
@@ -442,6 +443,80 @@ public void upsertObjs(@Nonnull Obj[] objs) throws ObjTooLargeException {
     }
   }
 
+  @Override
+  public boolean deleteConditional(@Nonnull UpdateableObj obj) {
+    ObjId id = obj.id();
+    Lock l = repo.objLock(id);
+    try {
+      RocksDBBackend b = backend;
+      TransactionDB db = b.db();
+      ColumnFamilyHandle cf = b.objs();
+      byte[] key = dbKey(id);
+
+      byte[] bytes = db.get(cf, key);
+      if (bytes == null) {
+        return false;
+      }
+      Obj existing = deserializeObj(id, bytes);
+      if (!existing.type().equals(existing.type())) {
+        return false;
+      }
+      UpdateableObj ex = (UpdateableObj) existing;
+      if (!ex.versionToken().equals(obj.versionToken())) {
+        return false;
+      }
+
+      db.delete(cf, key);
+      return true;
+    } catch (RocksDBException e) {
+      throw rocksDbException(e);
+    } finally {
+      l.unlock();
+    }
+  }
+
+  @Override
+  public boolean updateConditional(@Nonnull UpdateableObj expected, @Nonnull UpdateableObj newValue)
+      throws ObjTooLargeException {
+    ObjId id = expected.id();
+    checkArgument(id != null && id.equals(newValue.id()));
+    checkArgument(expected.type().equals(newValue.type()));
+    checkArgument(!expected.versionToken().equals(newValue.versionToken()));
+
+    Lock l = repo.objLock(id);
+    try {
+      RocksDBBackend b = backend;
+      TransactionDB db = b.db();
+      ColumnFamilyHandle cf = b.objs();
+      byte[] key = dbKey(id);
+
+      byte[] obj = db.get(cf, key);
+      if (obj == null) {
+        return false;
+      }
+      Obj existing = deserializeObj(id, obj);
+      if (!existing.type().equals(existing.type())) {
```

Comment:
```suggestion
      if (!existing.type().equals(expected.type())) {
```

Reply:
meh - the old code had that check - just the `if` was dumb